### PR TITLE
Add scheduler version enforcement

### PR DESCRIPTION
**Pull request by dadgar**
@armon For review

### CODE REVIEW

File: nomad/structs/structs.go [hunk 1]
```diff
@@ -305,8 +305,9 @@ type EvalAckRequest struct {
 
 // EvalDequeueRequest is used when we want to dequeue an evaluation
 type EvalDequeueRequest struct {
-	Schedulers []string
-	Timeout    time.Duration
+	Schedulers       []string
+	Timeout          time.Duration
+	SchedulerVersion uint16
```
Comment (armon) [inline nomad/structs/structs.go]: Something about the `uint16` feels dirty...

File: nomad/worker.go [hunk 2]
```diff
@@ -134,8 +134,9 @@ func (w *Worker) run() {
 func (w *Worker) dequeueEvaluation(timeout time.Duration) (*structs.Evaluation, string, bool) {
 	// Setup the request
 	req := structs.EvalDequeueRequest{
```
Comment (armon) [inline nomad/worker.go]: The worker should probably not sit in a hot loop of calling dequeue on this error, e.g. the failure is basically semi-permanent.

File: nomad/structs/structs.go (hunk 1 above)
Comment (dadgar) [inline nomad/structs/structs.go]: It gives us a lot of versions while keeping it low overhead. We will never pass 65536 version bumps.

File: nomad/worker.go (hunk 2 above)
Comment (dadgar) [inline nomad/worker.go]: So I was thinking we want it to spam the logs so the operator knows they are doing something bad. I agree that we should detect and start tuning back tho. But I don't think we should do anything permanent. Thoughts?

**Pull request merged**